### PR TITLE
Make web more resilient to garden being `stuck`

### PR DESCRIPTION
**Pull request by xtreme-sameer-vohra**
This addresses #3928 
- Adds a 5 min static request timeout to the StreamHijacker -> Stream function
- Adds a ctx to StreamHijacker -> Hijack function. This is utilized by `Container.Attach` & `Container.Run`

To be able to extend the container interface, this PR also pulls in the `garden/client` code.

### CODE REVIEW

File: atc/worker/gclient/client_stream_timeout_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	Context("blocking forever", func() {
+		BeforeEach(func() {
+			gServer.Reset()
+			gServer.AppendHandlers(func(w http.ResponseWriter, r *http.Request) {
+				<- ctx.Done()
+			})
+		})
+
+		It("fails once context expires", func() {
```
Comment (kcmannem) [inline atc/worker/gclient/client_stream_timeout_test.go]: very good test

File: atc/worker/gclient/connection/connection.go [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.streamProcess(ctx, handle, processIO, hijackedConn, hijackedResponseReader)
+}
+
+
+// TODO: add cancel func stuff
```
Comment (kcmannem) [inline atc/worker/gclient/connection/connection.go]: remove

File: atc/worker/worker.go [hunk 3]
```diff
@@ -631,6 +632,7 @@ func (worker *gardenWorker) cloneRemoteVolumes(
 }
 
 func (worker *gardenWorker) FindContainerByHandle(logger lager.Logger, teamID int, handle string) (Container, bool, error) {
+	// TODO: does this require a real context?
```
Comment (kcmannem) [inline atc/worker/worker.go]: remove

**Comment by xtreme-sameer-vohra** [comment]
@ddadlani DCO is set and branch is rebased with the task step changes.

**Pull request merged**